assets/js/components/OverlayNotification/LinkAnalyticsAndAdSenseAccountsOverlayNotification.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * LinkAnalyticsAndAdSenseAccountsOverlayNotification component.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { Button } from 'googlesitekit-components';
+import Data from 'googlesitekit-data';
+import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
+import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import { CORE_UI } from '../../googlesitekit/datastore/ui/constants';
+import { CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
+import { MODULES_ANALYTICS_4 } from '../../modules/analytics-4/datastore/constants';
+import AnalyticsAdsenseConnectGraphic from '../../../svg/graphics/analytics-adsense-connect.svg';
+import OverlayNotification from './OverlayNotification';
+import useViewOnly from '../../hooks/useViewOnly';
+import { isFeatureEnabled } from '../../features';
+
+const { useSelect, useDispatch } = Data;
+
+export const LINK_ANALYTICS_ADSENSE_OVERLAY_NOTIFICATION =
+	'LinkAnalyticsAndAdSenseAccountsOverlayNotification';
+
+export default function LinkAnalyticsAndAdSenseAccountsOverlayNotification() {
+	const ga4AdSenseIntegration = isFeatureEnabled( 'ga4AdSenseIntegration' );
+
+	const isViewOnly = useViewOnly();
+
+	const isShowingNotification = useSelect( ( select ) =>
+		select( CORE_UI ).isShowingOverlayNotification(
+			LINK_ANALYTICS_ADSENSE_OVERLAY_NOTIFICATION
+		)
+	);
+
+	const { setOverlayNotificationToShow, dismissOverlayNotification } =
+		useDispatch( CORE_UI );
+
+	const supportURL = useSelect( ( select ) =>
+		select( CORE_SITE ).getGoogleSupportURL( {
+			path: '/adsense/answer/6084409',
+		} )
+	);
+
+	const isDismissed = useSelect( ( select ) =>
+		select( CORE_USER ).isItemDismissed(
+			LINK_ANALYTICS_ADSENSE_OVERLAY_NOTIFICATION
+		)
+	);
+
+	const isDismissing = useSelect( ( select ) =>
+		select( CORE_USER ).isDismissingItem(
+			LINK_ANALYTICS_ADSENSE_OVERLAY_NOTIFICATION
+		)
+	);
+
+	const analyticsModuleConnected = useSelect( ( select ) => {
+		if ( isViewOnly || isDismissed ) {
```

Comment:
It prevents the selector (and associated resolver) from running when not needed. It's useful in tests (in part, we don't have to mock out as much stuff when this isn't needed/used), but also a good habit to get into for a component that doesn't always need to call selectors with resolvers.

Reply:
That's a good point. However, in this case, the `isItemDismissed` will return `undefined` while resolving, resulting in this guard to not take effect and the select function being called, at least initially. I guess it will prevent the select calls on subsequent renders once the `isDismissed` is resolved, by that point the resolvers related to the `select` call will run anyways. But I wonder if it's worth the mental overhead here. 


Good point about tests, I'll keep it as is for now. :+1: